Material.Ripple/RippleEffect.cs

```diff
@@ -110,6 +120,25 @@ private void RemoveLastRipple() {
             _last = null;
         }
 
+        public void RaiseRipple(double nX = 0.5, double nY = 0.5) {
+            var c = _container;
+
+            if (c is null || nX < 0 || nX > 1 || nY < 0 || nY > 1)
+                return;
+
+            lock (this) {
```

Comment:
1. lock (this) is antipattern, you should create a new object for this or lock to something private.
2. There is no locks in current pointer pressed handlers, this is intentional?

Reply:
yea i was concerning this API usage in other threads since this ripple effect implementation have things like `_pointer++`, `_last` and etc.

Reply:
So you probably should add locks to other places like pointer pressed handler